### PR TITLE
CA1311 various fixes

### PR DESCRIPTION
**Pull request by TKharaishvili**
This PR is a follow-up of these two - https://github.com/dotnet/roslyn-analyzers/pull/5970/ https://github.com/dotnet/roslyn-analyzers/pull/2186

@buyaa-n @Youssef1313 Here's the new PR addressing the issues left over from the previous ones.
I've updated the fixer to address the VB edge case and I've also fixed the typo in the resource files.
I haven't however moved unit tests into a single file. Is that a must? I see that many analyzers/fixers in the codebase have separate files for unit tests.

### CODE REVIEW

File: src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.vb [hunk 1]
```diff
@@ -11,15 +11,21 @@ Namespace Microsoft.NetCore.VisualBasic.Analyzers.Runtime
...
+            Debug.Assert(node.IsKind(SyntaxKind.InvocationExpression) Or node.IsKind(SyntaxKind.SimpleMemberAccessExpression))
 
-            Dim invocation = CType(invocationNode, InvocationExpressionSyntax)
-            If invocation.Expression.IsKind(SyntaxKind.SimpleMemberAccessExpression) Then
-                Return DirectCast(invocation.Expression, MemberAccessExpressionSyntax).Name.GetLocation()
+            If node.IsKind(SyntaxKind.InvocationExpression) Then
+                Dim invocation = DirectCast(node, InvocationExpressionSyntax)
+                If invocation.Expression.IsKind(SyntaxKind.SimpleMemberAccessExpression) Then
+                    Return DirectCast(invocation.Expression, MemberAccessExpressionSyntax).Name.GetLocation()
+                End If
```
Comment (Youssef1313) [inline src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.vb:21]:
```suggestion
                node = DirectCast(node, InvocationExpressionSyntax).Expression
```

File: src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.vb [hunk 2]
```diff
@@ -11,15 +11,21 @@ Namespace Microsoft.NetCore.VisualBasic.Analyzers.Runtime
     Public Class BasicSpecifyCultureForToLowerAndToUpperAnalyzer
         Inherits SpecifyCultureForToLowerAndToUpperAnalyzer
 
-        Protected Overrides Function GetMethodNameLocation(invocationNode As SyntaxNode) As Location
-            Debug.Assert(invocationNode.IsKind(SyntaxKind.InvocationExpression))
+        Protected Overrides Function GetMethodNameLocation(node As SyntaxNode) As Location
+            Debug.Assert(node.IsKind(SyntaxKind.InvocationExpression) Or node.IsKind(SyntaxKind.SimpleMemberAccessExpression))
```
Comment (Youssef1313) [inline src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.vb:15]:
```suggestion
            Debug.Assert(node.IsKind(SyntaxKind.InvocationExpression) OrElse node.IsKind(SyntaxKind.SimpleMemberAccessExpression))
```

File: src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.Fixer.vb [hunk 3]
```diff
@@ -21,25 +21,41 @@ Namespace Microsoft.NetCore.VisualBasic.Analyzers.Runtime
...
-                    Dim symbolInfo = model.GetSymbolInfo(node, cancellationToken).Symbol
-                    Dim methodSymbol = TryCast(symbolInfo, IMethodSymbol)
-
-                    If methodSymbol IsNot Nothing And methodSymbol.Parameters.Length = 0 Then
-                        Dim newArg = generator.Argument(CreateCurrentCultureMemberAccess(generator, model)).WithAdditionalAnnotations(Formatter.Annotation)
-                        Dim newInvocation = invocation.AddArgumentListArguments(DirectCast(newArg, ArgumentSyntax)).WithAdditionalAnnotations(Formatter.Annotation)
-                        Dim newRoot = root.ReplaceNode(invocation, newInvocation)
-                        Return document.WithSyntaxRoot(newRoot)
-                    End If
+            If ShouldFix(node) Then
```
Comment (buyaa-n) [inline src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.Fixer.vb:24]: Is this check needed when the parent already checking this before adding these fixes?
https://github.com/dotnet/roslyn-analyzers/blob/04edea1ac503805a8d9a6959ec18e20bff506bb5/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyCultureForToLowerAndToUpper.Fixer.cs#L24

Comment (TKharaishvili) [reply src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.Fixer.vb:24]: @buyaa-n as far as I've seen in the implementation of other fixers, they mirror some of the checks made in the analyzer. The types here don't guarantee that nodes are exactly the type you expect so defensive coding is a good idea I think.

Comment (buyaa-n) [reply src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.Fixer.vb:24]: I don't mind defensive code, but it looks to me that this method is just called twice, though I might not seeing all scenarios

File: src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.vb (hunk 1 above)
Comment (buyaa-n) [reply src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.vb:21]: Not sure why @Youssef1313  suggested this

**Review by buyaa-n — APPROVED** [review]
LGTM, thanks for the fix @TKharaishvili  and thanks for pointing it out @Youssef1313

File: src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.Fixer.vb [hunk 4]
```diff
@@ -21,25 +21,41 @@ Namespace Microsoft.NetCore.VisualBasic.Analyzers.Runtime
...
+                Dim memberAccess = DirectCast(node.Parent, MemberAccessExpressionSyntax)
+
+                If memberAccess.Parent Is Nothing Or Not memberAccess.Parent.IsKind(SyntaxKind.InvocationExpression) Then
```
Comment (TKharaishvili) [inline src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.Fixer.vb:27]: @buyaa-n I think this needs to be `OrElse` too or there's a risk of null-ref exception.
I'm sorry I wasn't able to point this out in time until the branch was merged.

Comment (Youssef1313) [reply src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.Fixer.vb:27]: @TKharaishvili there will be no null-ref exception. `IsKind` is an extension method and returns `false` if the receiver is null.

https://github.com/dotnet/roslyn/blob/9c4920a8d2538abd92d7e8880f556f75ad8a6588/src/Compilers/CSharp/Portable/CSharpExtensions.cs#L47-L50

Comment (TKharaishvili) [reply src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.Fixer.vb:27]: @Youssef1313 I see. Thanks for clarifying that.

Comment (buyaa-n) [reply src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.Fixer.vb:27]: Thanks for pointing this out, that makes sense, no worries you can always create a new PR for fix

Comment (buyaa-n) [reply src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.Fixer.vb:27]: Oh just saw @Youssef1313's reply, thanks clarification!
